integration/spark/spark3/src/main/java/io/openlineage/spark3/agent/lifecycle/plan/catalog/IcebergHandler.java

```diff
@@ -116,11 +116,11 @@ public Optional<TableProviderFacet> getTableProviderFacet(Map<String, String> pr
   @Override
   public Optional<String> getDatasetVersion(
       TableCatalog tableCatalog, Identifier identifier, Map<String, String> properties) {
-    SparkCatalog sparkCatalog = (SparkCatalog) tableCatalog;
     SparkTable table;
     try {
-      table = sparkCatalog.loadTable(identifier);
-    } catch (NoSuchTableException ex) {
+      table = ((SparkCatalog) tableCatalog).loadTable(identifier);
+    } catch (NoSuchTableException | ClassCastException e) {
+      log.error("Failed to load table from catalog: {}", identifier, e);
```

Comment:
Based on the Iceberg source, I think you can get away with casting the result of `loadTable` rather than casting the `tableCatalog` to a `SparkCatalog`. The `SparkSessionCatalog`'s internal `icebergCatalog` is [actually just a `SparkCatalog`](https://github.com/apache/iceberg/blob/apache-iceberg-0.13.1/spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkSessionCatalog.java#L69-L71) and its `loadTable` method [just delegates to the internal `icebergCatalog`](https://github.com/apache/iceberg/blob/apache-iceberg-0.13.1/spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkSessionCatalog.java#L116-L122). I think you can keep the `ClassCastException, but change this try block to 
```java
table = (SparkTable) tableCatalog.loadTable(identifier);
```

Reply:
Agreed. Given the underlying catalog is just `SparkCatalog`, casting the result of `SparkCatalog.loadTable()` is cleaner.